src/microbe_stage/editor/CellEditorComponent.GUI.cs

```diff
@@ -95,7 +95,7 @@ private void SetMembraneTooltips(MembraneType referenceMembrane)
     /// <summary>
     ///   Updates the fluidity / rigidity slider tooltip
     /// </summary>
-    private void SetRigiditySliderTooltip(int rigidity)
+    private void SetRigiditySliderTooltip(float rigidity)
```

Comment:
Shouldn't rigidity be kept as an integer? That's an unrelated change to switching to MP being a float.

Reply:
The signal actually binds a float and with the change to float MP might as well just use float here for consistency.

Reply:
I did notice quite a few places where it was also changed. I think for logical correctness enforcing that the rigidity slider value is a full int at certain points would make a lot of sense, I don't want there to be potential for bugs where partial steps are applied in some situations but then the slider can't go to the exact value due to the step size set in the GUI and the display gets out of sync with reality.

Reply:
Changing rigidity slider's logic back to int brings back #4150 in full force, where to go about this?

Reply:
I think someone needs to take a look at the math to find out why trying to create the rigidity step at no MP causes the divide by zero. I would guess that the reason it works with a float is that it ends up with a really small number instead of 0 and then all the math rounds out in display. But that isn't really desirable to me, I'd like the rigidity math to be fixed / behaviour fixed so that it just skips allowing the change to proceed when the incorrect condition is triggered.

Reply:
So apparently, when the cost multiplier is low enough (like 0.2), the cost per step for rigidity is rounded down to 0 and this is the source of the divide by zero problem.  Now my most logical solution in this circumstance would be to cap the rigidity's minimum cost to 1.

Reply:
That makes quite a lot of sense, though maybe with the fractional MP now being visible limiting it down to 0.5 or such a low value, would also work. I think a consistent "cap minimum price" method should be created so all parts of the editor work the same way when all game setting and multicellularity discounts are applied which could result in various things being very cheap.